### PR TITLE
fix(cli): preserve proxy env for host daemon

### LINKED ISSUE #1022: Behind a corporate proxy, the host daemon can't reach the model backend (proxy env vars stripped from its allowlist) — sharing what I hit + a local patch, looking for the right approach
### What I ran into

I'm trying out omnigent on macOS inside a corporate network that requires an authenticated HTTP proxy. On open WiFi everything works; behind the corporate proxy `omnigent run` reliably fails — the background **host daemon** never comes online and the CLI aborts after 30s with *"the connect daemon for host '…' did not come online within 30s."*

I dug into it and wanted to share what I found, plus the small patch I'm running locally — but I'm really opening this to **ask what the right approach is**, since it touches a deliberate design decision (the env allowlists) rather than just being an obvious bug.

### Environment

- omnigent 0.2.0, macOS
- Model backend via the Databricks provider
- Authenticated corporate proxy in the shell (`HTTPS_PROXY=http://USER:***@proxy.example.com:3128`)

### What's happening (as far as I can tell)

1. `omnigent run` spawns the background host daemon, which authenticates to the Databricks backend via `databricks-sdk`.
2. The daemon env is built by `_build_host_daemon_env` (`cli.py`). The standard proxy vars (`HTTP_PROXY`/`HTTPS_PROXY`/`NO_PROXY`/`ALL_PROXY`) aren't on its allowlists and match none of its prefixes (`LC_`, `MLFLOW_`, `OTEL_`, `ANTHROPIC_DEFAULT_`, `AZURE_OPENAI_`, `DATABRICKS_`, `OMNIGENT_`, `OPENAI_`), so they're dropped. (`OPENAI_PROXY` happens to survive via the `OPENAI_` prefix, but the Databricks SDK doesn't read it.)
3. With no `HTTPS_PROXY`, the SDK attempts a **direct** connection, which my corporate firewall silently blackholes → it hangs and times out after **5 minutes**, well past the 30s the CLI waits.

### Evidence

- `curl` to the Databricks workspace **through the proxy** → HTTP 200 in ~0.2s; the **same request direct** → times out (firewall blackhole).
- Daemon log: *"Failed to fetch host metadata from https://<workspace>/.well-known/databricks-config: Timed out after 0:05:00."*
- The running daemon's env confirms `HTTPS_PROXY` is absent.

### What I patched locally (to unblock myself)

I added the proxy vars to the two allowlists and it works end-to-end — the daemon connects through the proxy and a one-shot `run` reaches the model and replies normally:

- `cli.py` → `_LOCAL_DAEMON_ENV_ALLOWLIST`: added `HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY`, `NO_PROXY` (+ lowercase)
- `host/connect.py` → `_RUNNER_ENV_ALLOWLIST`: same additions

### Is there already a config-only workaround? (mostly, but not for the daemon)

Digging further, the picture is:

- In **local mode** the daemon env (`_build_host_daemon_env`) is built from the union of `_LOCAL_DAEMON_ENV_ALLOWLIST`, `_RUNNER_ENV_ALLOWLIST`, and their prefixes — all **static** — so adding the proxy vars to *either* allowlist fixes the local daemon.
- The **runner** additionally honors an opt-in passthrough, `OMNIGENT_RUNNER_ENV_PASSTHROUGH`, so on the runner side a user can forward proxy vars today with no code change.
- But that passthrough is read only in `_build_runner_env` — it's **not referenced in `cli.py`**, so it never reaches the daemon. There's no `OMNIGENT_DAEMON_ENV_PASSTHROUGH` equivalent, and no prefix a standard proxy var would match. **So there is no configuration-only way to get proxy vars into the daemon** — and since the daemon is what fails first, that's the actual gap.

### The open question (this is what I'd love your view on)

For my setup the runner is purely local, so forwarding proxy vars is harmless. But I recognize:

- proxy URLs can embed credentials (`user:pass@host`), and
- the runner allowlist is deliberately strict (spec self-containment — agents shouldn't inherit the host owner's shell secrets).

So **whether proxy vars should be forwarded by default — and to which of the daemon/runner — is your call, not mine.** Two shapes I could imagine, if you decide it's worth addressing:

1. **A daemon-side passthrough var** mirroring `OMNIGENT_RUNNER_ENV_PASSTHROUGH` (e.g. `OMNIGENT_DAEMON_ENV_PASSTHROUGH`) — keeps forwarding **opt-in**, changes no defaults, and reuses a pattern that already exists.
2. **Adding the proxy vars to `_LOCAL_DAEMON_ENV_ALLOWLIST` directly** (daemon-only) — smallest change, fixes the failure by default. There's precedent: `pi_executor.py`, `codex_executor.py`, and `codex_native_app_server.py` already forward these to their spawned processes.

Mainly wanted to surface the failure mode and the diagnosis so you can decide whether this is something worth handling and which approach fits the design best. Thanks!

### PR DESCRIPTION
**Pull request by roian6**
## Related issue

Closes #1022

## Summary

Corporate-proxy users could not bring the background host daemon online because
its allowlisted environment dropped the standard proxy selectors before the
daemon contacted a model backend.

- preserve `HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY`, `NO_PROXY`, and lowercase
  variants in both local and remote host-daemon modes
- keep proxy variables out of runner subprocesses by default; the existing
  `OMNIGENT_RUNNER_ENV_PASSTHROUGH` opt-in remains unchanged
- add focused coverage for both daemon modes, uppercase/lowercase names, the
  runner boundary, and the local/remote provider-secret split

ELI5: the trusted host daemon can use the operator's corporate proxy, while an
agent runner still receives proxy credentials only when the operator explicitly
opts in.

```text
operator shell ── proxy env ──> host daemon ── filtered env ──> runner
                               retained                     excluded by default
```

## Test Plan

- `OMNIGENT_SKIP_WEB_UI=true uv run pytest tests/cli/test_host_daemon_env.py -q`
  - RED on current `main`: `2 failed, 2 passed` because both daemon modes
    omitted all eight proxy names while the runner-boundary cases passed
  - GREEN after the fix: `4 passed`
- focused existing daemon/runner regression selection: `11 passed`
- controlled helper matrix using `uv run python`: both daemon modes retained all
  eight names; runner default retained none; explicit passthrough retained only
  the two named proxies; local/remote provider-secret behavior remained split
- `OMNIGENT_SKIP_WEB_UI=true uv run ruff check omnigent/cli.py tests/cli/test_host_daemon_env.py`
- `OMNIGENT_SKIP_WEB_UI=true uv run ruff format --check omnigent/cli.py tests/cli/test_host_daemon_env.py`
- `OMNIGENT_SKIP_WEB_UI=true uv run pre-commit run --files omnigent/cli.py tests/cli/test_host_daemon_env.py --show-diff-on-failure`
- `git diff --check`

`pre-commit run --all-files` also passed every backend and repository-hygiene
hook, but could not run the unrelated `web-prettier` hook because npm is not
installed in the verification environment. No `web/` file changed.

## Demo

N/A — non-visual daemon environment fix.

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] UI / frontend change
- [ ] Refactor / chore
- [ ] Docs
- [x] Test / CI
- [ ] Breaking change

## Test coverage

- [x] Unit tests added / updated
- [ ] Integration tests added / updated
- [ ] E2E tests added / updated
- [ ] Manual verification completed
- [ ] Existing tests cover this change
- [ ] Not applicable

## Coverage notes

The deterministic environment-builder seam covers the bug without requiring a
live corporate proxy or model backend. A controlled helper probe exercised the
real daemon and runner environment builders with synthetic values and asserted
the full boundary matrix. This bug fix does not introduce new user-facing
functionality, so no new e2e test is required.

## Changelog

Host daemons now honor standard proxy environment variables without forwarding
them to runners by default.

### CODE REVIEW

**Comment by roian6** [comment]
Updated this branch onto current `main` and reran the focused backend checks.

Validation:
- `uv run pytest tests/cli/test_cli.py::test_host_daemon_env_preserves_standard_proxy_vars -q` — 1 passed
- `uv run ruff check omnigent/cli.py tests/cli/test_cli.py` — passed
- `uv run ruff format --check omnigent/cli.py tests/cli/test_cli.py` — already formatted

**Comment by roian6** [comment]
Reworked this PR onto current `main` at `3f53c90f982aee9e88dbadbdf4aad4754b9ce854` rather than mechanically carrying forward the stale test location.

Validation on the new head:

- current-main RED: `2 failed, 2 passed` (both daemon modes stripped all eight proxy names)
- focused GREEN: `4 passed`
- adjacent daemon/runner suites: `169 passed`
- Ruff check/format, touched-file pre-commit, `git diff --check`, CRLF scan, and added-line secret scan: passed

The patch preserves the eight standard uppercase/lowercase proxy variables for both host-daemon modes. Runner forwarding remains excluded by default and still requires the existing explicit passthrough. The PR body has been refreshed to the current template and commands.

**Comment by roian6** [comment]
CI follow-up: the refreshed head completed with all code-adjacent checks green. Three of four E2E shards passed; shard 1 failed only in the existing `test_repl_approve_always_caches_for_later_turns` UI timing assertion because the second-turn `auto-approved` audit text was not captured. This PR changes only the host-daemon proxy allowlist and its unit test.

I attempted a failed-job rerun, but GitHub does not permit this fork contributor to rerun the upstream workflow. Could a maintainer rerun failed job `89745710837` (run `30184116209`) when convenient?

File: omnigent/cli.py [hunk 1]
```diff
@@ -2648,7 +2661,9 @@ def _build_host_daemon_env(
         env = {
             key: value
             for key, value in os.environ.items()
-            if key in _RUNNER_ENV_ALLOWLIST or key.startswith(daemon_env_prefixes)
+            if key in _RUNNER_ENV_ALLOWLIST
+            or key in _HOST_DAEMON_PROXY_ENV_ALLOWLIST
```
[the remark this answers is not in the document]
Comment (roian6) [reply omnigent/cli.py:3049]: Good catch — updated the comment in `34513db78b16b918f1a72d48d8b36393ced251c9` to explicitly include standard proxy selectors while retaining the provider-secret boundary. Re-verified the 4 focused regression cases, Ruff check/format, touched-file pre-commit, and `git diff --check`.

**Comment by roian6** [comment]
Current-head CI follow-up for `34513db78b16b918f1a72d48d8b36393ced251c9`:

Run [30184679118](https://github.com/omnigent-ai/omnigent/actions/runs/30184679118) is green across the code-adjacent test matrix except [`Pytest (stores-postgres)` job 89747153922](https://github.com/omnigent-ai/omnigent/actions/runs/30184679118/job/89747153922). That job failed during **Initialize containers**, before checkout, dependency installation, or any tests ran. The runner attempted `docker pull postgres:16` three times; each attempt timed out against Docker Hub (`Client.Timeout exceeded while awaiting headers` / `context deadline exceeded`). The sibling `stores-mysql` and `stores` jobs passed, as did all four E2E shards and all three E2E UI shards.

Could a maintainer please rerun the failed `stores-postgres` job for this current head when convenient? I am not pushing a no-op commit for this external registry timeout. The separate `Maintainer Approval` failure is the expected policy gate.

**Review by dhruv0811 — APPROVED** [review]
LGTM! Sorry for the very late review on this pr, thanks for your contribution!
